components/nimbus/ios/scripts/nimbus-fml.sh

```diff
@@ -101,16 +101,24 @@ find_as_version() {
         # We try to resolve that, and find the version from the Package.swift file in that local directory.
         # https://github.com/mozilla-mobile/firefox-ios/issues/12243
         rust_components_path=$(grep -A 3 $'XCRemoteSwiftPackageReference "rust-components-swift"' "$SOURCE_ROOT/$PROJECT.xcodeproj/project.pbxproj" | grep 'repositoryURL = "file://' | grep -o -E '/\w[^"]+')
-        number_string=$(grep 'let version =' "$rust_components_path/Package.swift" | grep -E -o "\d+\.0.\d+")
+        number_string=$(grep 'let version =' "$rust_components_path/Package.swift" | grep -E -o "\d+\.\d+\.\d+")
     fi
 
     if [ -z "$number_string" ]; then
         echo "Error: No https://github.com/mozilla/rust-components-swift package was detected."
         echo "The package must be added as a project dependency."
         exit 2
     fi
-
-    AS_VERSION=${number_string//\.0\./\.} # rust-component-swift tags have a middle `.0.` to force it to align with spm. We remove it
+    # We try and differentiate between two version types.
+    # Nightlies: 121.0.20231027173509 --> 121.20231027173509
+    # Releases:  121.0.1 --> 121.0.1
+    local patch=${number_string##*.}
+    if [[ ${#patch} -gt 2 ]] ; then
+        # rust-component-swift tags have a middle `.0.` to force it to align with spm. We remove it
+        AS_VERSION=${number_string//\.0\./\.}
+    else
+        AS_VERSION=${number_string}
+    fi
```

Comment:
to clarify the issue, and this is where it's a little tricky, the rust-component-swift version  (`number_string`) we have here will always be 3 components. It's the AS_VERSION that **may** be two or three components. I can't think of an easy way to find out whether it's two or three components from here other than sending a network request and seeing if it's not a success

@bendk This makes me think we should think about policy a little, there are a few policy fixes for the future:
- If we make AS **always** be three or **always** be two components this will be simpler
- Maybe we could somehow decouple the version of the AS and the version of the FML?

Reply:
so this code maybe worth commenting.

The SPM has three components; this code checks that last component as the `patch` version (`${number_string##*.}`, then checks the length of the patch version component (`${#patch}`). If it's greater than 2, then treat it like a date stamp, and use it as a nightly. I figured that anything more than`121.0.99` would be _pathological_ as a release number.

Reply:
Ok, going to talk in slack, then document here.

Reply:
Decided to not to look for a forever solution until https://github.com/mozilla-mobile/firefox-ios/pull/17143 lands, then re-evaluate.

This PR should be good for then too.